### DISCUSSION TITLE
Avoid blank cells in a table, use N.A.

### DISCUSSION THREAD

**Discussion by kellijohnson-NOAA** [pfmc-assessments/sa4ss#87] [Show and tell]
When creating tables, **use a placeholder for blank cells rather than leaving them empty**. When some screen readers or sighted readers encounter empty cells within tables they may wonder if data was omitted by mistake. In addition, screen reader software may skip reading empty data cells, causing mismatches between header rows and their corresponding data cells.

## ❌bad example 
| Header 1  | Header 2            | Header 3  |
| --------- | ------------------- | --------- |
| Some data |                     | Cell 3    |
| data      | Some long data here | more data |

## a more detailed example
Below is a table of alternative ways to fill a blank cell in a table and what will be read by two screen readers given the table entry. The table was found on [section5oh8.com](http://section5oh8.com/pdf-remediation/how-to-handle-blank-data-cells-in-accessible-tables/).

|Alternative Values | JAWS | NVDA
|-- | -- | --
|n/a | En slash Uh | En slash A
|N.A. | N dot A | N dot A
|Na | Nah | En A
|  | blank | Nothing voiced
|Not applicable | Not applicable | Not applicable
|No value | No value | No value
|– | Dash | Nothing voiced
|0 | Zero | Zero

## 🏆 best practice
This appears to be determined but I would advocate for "N.A." because at least it is read commonly across screen readers, it clearly indicates that no value was placed in a cell, and it does not take up very much room, where "not applicable" could increase the width of your columns beyond that available.

**Comment by brianlangseth-NOAA** [comment]
@kellijohnson-NOAA Thanks for alerting us of this, I've always used "-" but see how that can cause a problem. I'll admit that N.A. looks weird to me. Is NA read the same as Na? Removing the periods looks cleaner, however if NA is read the same as Na, then I am content to have it be visually more busy to be non-visually more friendly.

**Comment by iantaylor-NOAA** [comment]
I looked through the tables in the 2021 lingcod north report and there were only a few with blank cells and a few that already had "NA". I'm fine changing "NA" to "N.A." where they occur. The table of tows, trips and input sample sizes had blanks and I think adding N.A. there is fine.

The one table where I would do something different is the one shown below, where I think the value could depend on the column. My suggestion would be to have the "Status" column show "fixed" rather than blank or N.A.  The "SD" column could have either 0 or N.A. for those fixed parameters, and the "Prior" column could have "none".
![image](https://user-images.githubusercontent.com/4992918/224386435-45f2d3f7-3007-47cb-aefa-ab1bbeafea81.png)

**Comment by kellijohnson-NOAA** [comment]
The most recent guidance that I was given is to use "empty cell" and shrink the font to something very small and change the color to be the same as the background color. This will be a bit complicated to code for us because of the desire to have either pdf or html output but would be really easy in power point presentations or if using just html. In the meantime, I am still going with "N.A." as the best option until we can implement hidden text.

FYI ... screen readers do not tell the user anything about font size or color so something that is not available to the sighted reader is just fine for the screen reader. Though, I also believe that @puntae has a preference of not leaving cells blank as well but I cannot remember. Maybe he just wanted what a blank cell meant defined in the caption.
